File: 4-implementation/knight.cpp
#include <stdio.h>

using namespace std;

int row, col, cnt;
int dx[8] = {-2, -2, -1, 1, 2, 2, 1, -1}, dy[8] = {-1, 1, 2, 2, 1, -1, -2, -2};

void countCase() {
    int destRow, destCol;

    for (int i = 0; i < 8; ++i) {
        destRow = row + dx[i];
        destCol = col + dy[i];

        if (destRow > 0 && destRow <= 8 && destCol > 0 && destCol <= 8) {
            ++cnt;
        }
    }
}

int main() {
    char input[3];

    scanf("%s", input);

    row = input[1] - '0';
    col = input[0] - 'a' + 1;

    countCase();

    printf("%d", cnt);

    return 0;
}

File: 4-implementation/knight.py
col, row = list(input())

# char를 int로 변환
row = int(row)
col = ord(col) - 96

count = 8

# row 변화량 배열
dr = [-1, 1, 2, 2, 1, -1, -2, -2]
# col 변화량 배열
dc = [2, 2, 1, -1, -2, - 2, - 1, 1]

for i in range(0, len(dr)):
    nr = row + dr[i]
    nc = col + dc[i]

    if (nr > 8 or nc > 8 or nr < 1 or nc < 1):
        count -= 1

print(count)
